src/io/mod.rs

```diff
@@ -313,6 +331,27 @@ mod tests {
         assert_ne!(text.metadata().unwrap().len(), gzipped.metadata().unwrap().len());
     }
 
+    #[test]
+    fn test_reading_and_writing_zstd_files() {
+        let lines = vec!["foo", "bar", "baz"];
+        let tempdir = TempDir::new().unwrap();
+        let text = tempdir.path().join("text.txt");
+        let zstd_compressed = tempdir.path().join("zstd_compressed.txt.zst");
+
+        let io = Io::default();
+        io.write_lines(&text, &mut lines.iter()).unwrap();
+        io.write_lines(&zstd_compressed, &mut lines.iter()).unwrap();
+
+        let r1 = io.read_lines(&text).unwrap();
+        let r2 = io.read_lines(&zstd_compressed).unwrap();
+
+        assert_eq!(r1, lines);
+        assert_eq!(r2, lines);
+
+        // Also check that we actually wrote zstd encoded data to the zstd file!
+        assert_ne!(text.metadata().unwrap().len(), zstd_compressed.metadata().unwrap().len());
```

Comment:
almost, it checks that the two files are different, not that the second file is zstd encoded!

Reply:
Perils of copy/paste vol. 487. Fixed.